src/utils/temporal.cpp

```diff
@@ -514,6 +515,275 @@ size_t LocalDateTimeHash::operator()(const LocalDateTime &local_date_time) const
   return result;
 }
 
+namespace {
+inline constexpr auto *kSupportedZonedDateTimeFormatsHelpMessage = R"help(
+A string representing ZonedDateTime should have the following structure:
+
+- <DateTime><timezone>
+
+The <DateTime> substring should use one of the following formats:
+
+- YYYY-MM-DDThh:mm:ss
+- YYYY-MM-DDThh:mm
+
+or
+
+- YYYYMMDDThhmmss
+- YYYYMMDDThhmm
+- YYYYMMDDThh
+
+The <timezone> substring should use one of the following formats:
+
+- Z
+- ±hh:mm
+- ±hh:mm[ZoneName]
+- ±hhmm
+- ±hhmm[ZoneName]
+- ±hh
+- ±hh[ZoneName]
+- [ZoneName]
+
+Symbol table:
+|---|---------|
+| Y | YEAR    |
+|---|---------|
+| M | MONTH   |
+|---|---------|
+| D | DAY     |
+|---|---------|
+| h | HOURS   |
+|---|---------|
+| m | MINUTES |
+|---|---------|
+| s | SECONDS |
+|---|---------|
+| Z | UTC     |
+|---|---------|
+
+ZoneName is a standardized timezone name from the IANA time zone database, given without quote marks.
+
+Additionally, seconds can be defined as decimal fractions with 3 or 6 digits after the decimal point.
+First 3 digits represent milliseconds, while the second 3 digits represent microseconds.
+
+It's important to note that the date and time parts should use both the corresponding separators
+or both parts should be written in their basic forms without the separators.)help";
+}  // namespace
+
+Timezone ParseTimezoneFromName(std::string_view timezone_string) {
+  auto extract_timezone_name = [](std::string_view &string) {
+    if (string.front() != '[' or string.back() != ']') {
```

Comment:
Instead of `front` and `back`, please prefer `std::basic_string_view<CharT,Traits>::starts_with` and `std::basic_string_view<CharT,Traits>::ends_with`, they are a bit more descriptive in this use-case.